homeassistant/components/media_player/denon.py

```diff
@@ -0,0 +1,192 @@
+      Only seeking via simulated button presses is possible.
+
+Configuration:
+
+To use your Denon you will need to add something like the following to
+your config/configuration.yaml:
+
+media_player:
+  platform: denon
+  name: Music station
+  host: 192.168.0.123
+
+Variables:
+
+host
+*Required
+The ip of the player. Example: 192.168.0.123
+
+name
+*Optional
+The name of the device.
+"""
+import telnetlib
+import logging
+
+from homeassistant.components.media_player import (
+    MediaPlayerDevice, SUPPORT_PAUSE, SUPPORT_VOLUME_SET,
+    SUPPORT_VOLUME_MUTE, SUPPORT_PREVIOUS_TRACK, SUPPORT_NEXT_TRACK,
+    SUPPORT_TURN_ON, SUPPORT_TURN_OFF,
+    DOMAIN)
+from homeassistant.const import (
+    CONF_HOST, STATE_OFF, STATE_IDLE, STATE_UNKNOWN)
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_DENON = SUPPORT_PAUSE | SUPPORT_VOLUME_SET | SUPPORT_VOLUME_MUTE | \
+    SUPPORT_PREVIOUS_TRACK | SUPPORT_NEXT_TRACK | \
+    SUPPORT_TURN_ON | SUPPORT_TURN_OFF
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the Denon platform. """
+    if not config.get(CONF_HOST):
+        _LOGGER.error(
+            "Missing required configuration items in %s: %s",
+            DOMAIN,
+            CONF_HOST)
+        return False
+
+    add_devices([
+        DenonDevice(
+            config.get('name', 'Music station'),
+            config.get('host'))
+    ])
+
+    return True
+
+
+class DenonDevice(MediaPlayerDevice):
+    """ Represents a Denon device. """
+
+    # pylint: disable=too-many-public-methods
+
+    def __init__(self, name, host):
+        self._name = name
+        self._host = host
+        self._telnet = telnetlib.Telnet(self._host)
+
+    def query(self, message):
+        """ Send request and await response from server """
+        try:
+            # unspecified command, should be ignored
+            self._telnet.write("?".encode('UTF-8') + b'\r')
+        except (EOFError, BrokenPipeError, ConnectionResetError):
+            self._telnet.open(self._host)
+
+        _LOGGER.error(self._telnet.read_very_eager())  # skip
+
+        _LOGGER.error(message)
+        self._telnet.write(message.encode('ASCII') + b'\r')
+        # timeout 200ms, defined by protocol
+        resp = self._telnet.read_until(b'\r', timeout=0.2)\
+            .decode('UTF-8').strip()
+
+        if message == "PW?":
+            # workaround; PW? sends also SISTATUS
+            _LOGGER.error(self._telnet.read_until(b'\r', timeout=0.2))
+        _LOGGER.error(resp)
+        return resp
+
+    @property
+    def name(self):
+        """ Returns the name of the device. """
+        return self._name
+
+    @property
+    def state(self):
+        """ Returns the state of the device. """
+        pwstate = self.query('PW?')
+        if pwstate == "PWSTANDBY":
+            return STATE_OFF
+        if pwstate == "PWON":
+            return STATE_IDLE
```

Comment:
It looks like their API doesn't allow you to determine if the receiver is playing or not. You may want this to return STATE_ON instead of STATE_IDLE.